libs/components/lists/src/lib/modules/repeater/repeater.component.spec.ts

```diff
@@ -133,6 +133,52 @@ describe('Repeater item component', () => {
     flushDropdownTimer();
   }));
 
+  it('should update the repeater service items and their order items dynamically changed', fakeAsync(() => {
+    const fixture = TestBed.createComponent(RepeaterTestComponent);
+    const cmp: RepeaterTestComponent = fixture.componentInstance;
+    cmp.showRepeaterWithNgFor = true;
+    detectChangesAndTick(fixture);
+
+    const repeaterSvc = fixture.debugElement
+      .query(By.css('sky-repeater'))
+      .injector.get(SkyRepeaterService);
+    let repeaterItems = cmp.repeater?.items?.toArray();
```

Comment:
The block of code that gets the repeater items and validates their properties is almost identical to the block of code at the end of the test. Create a function that encapsulates this logic.

```
function validateRepeaterItemOrder(fixture: ComponentFixture<RepeaterTestComponent>, firstItemTag: string): void {
    const cmp = fixture.componentInstance;
    const repeaterSvc = fixture.debugElement
      .query(By.css('sky-repeater'))
      .injector.get(SkyRepeaterService);

    const repeaterItems = cmp.repeater?.items?.toArray();

    expect(repeaterItems).not.toBeUndefined();

    if (repeaterItems) {
      expect(repeaterSvc.items).toEqual(repeaterItems);
      expect(repeaterItems[0].tag).toEqual(firstItemTag);
      expect(repeaterSvc.items[0].tag).toEqual(firstItemTag);
    }
}
```

Then your test becomes small and easy to read:

```
it('should update the repeater service items and their order items dynamically changed', fakeAsync(() => {
    const fixture = TestBed.createComponent(RepeaterTestComponent);
    const cmp = fixture.componentInstance;

    cmp.showRepeaterWithNgFor = true;
    detectChangesAndTick(fixture);

    validateRepeaterItemOrder(fixture, 'item1');

    cmp.items = [
      {
        id: 'item3',
        title: 'Item 3',
      },
      {
        id: 'item2',
        title: 'Item 2',
      },
      {
        id: 'item1',
        title: 'Item 1',
      },
    ];

    detectChangesAndTick(fixture);

    validateRepeaterItemOrder(fixture, 'item3');
}));
```